subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/ExecuteActionsTaskExecuter.java

```diff
@@ -241,6 +241,16 @@ public boolean isAllowOverlappingOutputs() {
             return true;
         }
 
+        @Override
+        public boolean isOverlappingOutputsDetected() {
+            return context.getOverlappingOutputs().isPresent();
+        }
+
+        @Override
+        public boolean isCleanupOutputsOnNonIncrementalExecution() {
+            return isRequiresInputChanges() && !isRequiresLegacyInputChanges();
```

Comment:
We should change these two methods into one returning an enum.

Reply:
Returning an enum now.